### ISSUE TITLE
BUG: signal.ShortTimeFFT: `istft` with `mfft > len(win)` not supported with leading dimensions

### ISSUE THREAD

**Issue by nedlrichards** [scipy/scipy#22204]
### Describe your issue.

Thanks for this great tool! I am still trying to figure out all of it's functionality, and I really appreciate the comprehensive documentation

There seems to be support in `ShortTimeFFT` for `istft` when `mfft > len(win)` when the data is 1D, but not with 2D. I suspect the lack of a leading elipsis [here](https://github.com/scipy/scipy/blob/92d2a8592782ee19a1161d0bf3fc2241ba78bb63/scipy/signal/_short_time_fft.py#L1628C16-L1628C53).

### Reproducing Code Example

```python
import numpy as np
from scipy.signal import ShortTimeFFT
from scipy.signal.windows import tukey, hann

window_length = 201
hop_length = 150
fft_length = 300
num_samples = 2001
fs = 1000
num_channels = 8

# chirp
T = 2
f0 = 100
f1 = 200
c = (f1 - f0) / T
phase = (np.arange(num_channels) * np.pi / num_channels)
t_axis = np.arange(num_samples) / fs

# BUG: raises ValueError
phase = 2 * np.pi * (c * t_axis**2 / 2 + f0 * t_axis)[None, :] + phase[:, None]
# OK
#phase = 2 * np.pi * (c * t_axis**2 / 2 + f0 * t_axis)

time_series = tukey(num_samples, 0.1) * np.sin(phase)

window = hann(window_length)
short_time_fft = ShortTimeFFT(window, hop_length, fs, mfft=fft_length)
reference_stft = short_time_fft.stft(time_series)
test = short_time_fft.istft(reference_stft)
```

### Error message

```shell
File ~/miniconda3/lib/python3.11/site-packages/scipy/signal/_short_time_fft.py:1070, in ShortTimeFFT.istft(self, S, k0, k1, f_axis, t_axis)
   1067 x = np.zeros(S.shape[:-2] + (n_pts,),
   1068              dtype=float if self.onesided_fft else complex)
   1069 for q_ in range(q0, q1):
-> 1070     xs = self._ifft_func(S[..., :, q_ - self.p_min]) * self.dual_win
   1071     i0 = q_ * self.hop - self.m_num_mid
   1072     i1 = min(i0 + self.m_num, n_pts+k0)

ValueError: operands could not be broadcast together with shapes (8,300) (201,)
```

### SciPy/NumPy/Python version and system information

```shell
1.14.1 1.26.2 sys.version_info(major=3, minor=11, micro=5, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /home/ned.richards/miniconda3/include
    lib directory: /home/ned.richards/miniconda3/lib
    name: blas
    openblas configuration: unknown
    pc file directory: /home/ned.richards/miniconda3/lib/pkgconfig
    version: 3.9.0
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /home/ned.richards/miniconda3/include
    lib directory: /home/ned.richards/miniconda3/lib
    name: lapack
    openblas configuration: unknown
    pc file directory: /home/ned.richards/miniconda3/lib/pkgconfig
    version: 3.9.0
  pybind11:
    detection method: pkgconfig
    include directory: /home/ned.richards/miniconda3/include
    name: pybind11
    version: 2.13.6
Compilers:
  c:
    args: -march=nocona, -mtune=haswell, -ftree-vectorize, -fPIC, -fstack-protector-strong,
      -fno-plt, -O2, -ffunction-sections, -pipe, -isystem, /home/ned.richards/miniconda3/include,
      -fdebug-prefix-map=/home/conda/feedstock_root/build_artifacts/scipy-split_1733620549877/work=/usr/local/src/conda/scipy-split-1.14.1,
      -fdebug-prefix-map=/home/ned.richards/miniconda3=/usr/local/src/conda-prefix,
      -DNDEBUG, -D_FORTIFY_SOURCE=2, -O2, -isystem, /home/ned.richards/miniconda3/include
    commands: /home/conda/feedstock_root/build_artifacts/scipy-split_1733620549877/_build_env/bin/x86_64-conda-linux-gnu-cc
    linker: ld.bfd
    linker args: -Wl,-O2, -Wl,--sort-common, -Wl,--as-needed, -Wl,-z,relro, -Wl,-z,now,
      -Wl,--disable-new-dtags, -Wl,--gc-sections, -Wl,--allow-shlib-undefined, -Wl,-rpath,/home/ned.richards/miniconda3/lib,
      -Wl,-rpath-link,/home/ned.richards/miniconda3/lib, -L/home/ned.richards/miniconda3/lib,
      -march=nocona, -mtune=haswell, -ftree-vectorize, -fPIC, -fstack-protector-strong,
      -fno-plt, -O2, -ffunction-sections, -pipe, -isystem, /home/ned.richards/miniconda3/include,
      -fdebug-prefix-map=/home/conda/feedstock_root/build_artifacts/scipy-split_1733620549877/work=/usr/local/src/conda/scipy-split-1.14.1,
      -fdebug-prefix-map=/home/ned.richards/miniconda3=/usr/local/src/conda-prefix,
      -DNDEBUG, -D_FORTIFY_SOURCE=2, -O2, -isystem, /home/ned.richards/miniconda3/include
    name: gcc
    version: 13.3.0
  c++:
    args: -fvisibility-inlines-hidden, -fmessage-length=0, -march=nocona, -mtune=haswell,
      -ftree-vectorize, -fPIC, -fstack-protector-strong, -fno-plt, -O2, -ffunction-sections,
      -pipe, -isystem, /home/ned.richards/miniconda3/include, -fdebug-prefix-map=/home/conda/feedstock_root/build_artifacts/scipy-split_1733620549877/work=/usr/local/src/conda/scipy-split-1.14.1,
      -fdebug-prefix-map=/home/ned.richards/miniconda3=/usr/local/src/conda-prefix,
      -DNDEBUG, -D_FORTIFY_SOURCE=2, -O2, -isystem, /home/ned.richards/miniconda3/include
    commands: /home/conda/feedstock_root/build_artifacts/scipy-split_1733620549877/_build_env/bin/x86_64-conda-linux-gnu-c++
    linker: ld.bfd
    linker args: -Wl,-O2, -Wl,--sort-common, -Wl,--as-needed, -Wl,-z,relro, -Wl,-z,now,
      -Wl,--disable-new-dtags, -Wl,--gc-sections, -Wl,--allow-shlib-undefined, -Wl,-rpath,/home/ned.richards/miniconda3/lib,
      -Wl,-rpath-link,/home/ned.richards/miniconda3/lib, -L/home/ned.richards/miniconda3/lib,
      -fvisibility-inlines-hidden, -fmessage-length=0, -march=nocona, -mtune=haswell,
      -ftree-vectorize, -fPIC, -fstack-protector-strong, -fno-plt, -O2, -ffunction-sections,
      -pipe, -isystem, /home/ned.richards/miniconda3/include, -fdebug-prefix-map=/home/conda/feedstock_root/build_artifacts/scipy-split_1733620549877/work=/usr/local/src/conda/scipy-split-1.14.1,
      -fdebug-prefix-map=/home/ned.richards/miniconda3=/usr/local/src/conda-prefix,
      -DNDEBUG, -D_FORTIFY_SOURCE=2, -O2, -isystem, /home/ned.richards/miniconda3/include
    name: gcc
    version: 13.3.0
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.0.11
  fortran:
    args: -march=nocona, -mtune=haswell, -ftree-vectorize, -fPIC, -fstack-protector-strong,
      -fno-plt, -O2, -ffunction-sections, -pipe, -isystem, /home/ned.richards/miniconda3/include,
      -I/home/conda/feedstock_root/build_artifacts/scipy-split_1733620549877/_build_env/include,
      -fdebug-prefix-map=/home/conda/feedstock_root/build_artifacts/scipy-split_1733620549877/work=/usr/local/src/conda/scipy-split-1.14.1,
      -fdebug-prefix-map=/home/ned.richards/miniconda3=/usr/local/src/conda-prefix
    commands: /home/conda/feedstock_root/build_artifacts/scipy-split_1733620549877/_build_env/bin/x86_64-conda-linux-gnu-gfortran
    linker: ld.bfd
    linker args: -Wl,-O2, -Wl,--sort-common, -Wl,--as-needed, -Wl,-z,relro, -Wl,-z,now,
      -Wl,--disable-new-dtags, -Wl,--gc-sections, -Wl,--allow-shlib-undefined, -Wl,-rpath,/home/ned.richards/miniconda3/lib,
      -Wl,-rpath-link,/home/ned.richards/miniconda3/lib, -L/home/ned.richards/miniconda3/lib,
      -march=nocona, -mtune=haswell, -ftree-vectorize, -fPIC, -fstack-protector-strong,
      -fno-plt, -O2, -ffunction-sections, -pipe, -isystem, /home/ned.richards/miniconda3/include,
      -I/home/conda/feedstock_root/build_artifacts/scipy-split_1733620549877/_build_env/include,
      -fdebug-prefix-map=/home/conda/feedstock_root/build_artifacts/scipy-split_1733620549877/work=/usr/local/src/conda/scipy-split-1.14.1,
      -fdebug-prefix-map=/home/ned.richards/miniconda3=/usr/local/src/conda-prefix
    name: gcc
    version: 13.3.0
  pythran:
    include directory: ../../_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.11/site-packages/pythran
    version: 0.17.0
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /home/ned.richards/miniconda3/bin/python
  version: '3.11'
```

**Comment by lucascolley** [comment]
cc @DietBru

**Comment by DietBru** [comment]
It's always nice to receive a comprehensive bug report, pin-pointing the exact line of the problem (including a reproducible example of course).

This  problem is already fixed in PR #21673, which is included in the upcoming 1.15 release.
